src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -83,5 +83,229 @@ AS
     COMMIT TRANSACTION
 GO
 
+/***************************************************************************************/
+-- STORED PROCEDURE
+--    Reindex instance
+--
+-- DESCRIPTION
+--    Reidex instance
+--
+-- PARAMETERS
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @seriesInstanceUid
+--         * The series instance UID.
+--     @sopInstanceUid
+--         * The SOP instance UID.
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @longExtendedQueryTags
+--         * Long extended query tag data
+--     @doubleExtendedQueryTags
+--         * Double extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.ReindexInstance
+    @studyInstanceUid       VARCHAR(64),
+    @seriesInstanceUid      VARCHAR(64),
+    @sopInstanceUid         VARCHAR(64),               
+    @stringExtendedQueryTags dbo.InsertStringExtendedQueryTagTableType_1 READONLY,    
+    @longExtendedQueryTags dbo.InsertLongExtendedQueryTagTableType_1 READONLY,
+    @doubleExtendedQueryTags dbo.InsertDoubleExtendedQueryTagTableType_1 READONLY,
+    @dateTimeExtendedQueryTags dbo.InsertDateTimeExtendedQueryTagTableType_1 READONLY,
+    @personNameExtendedQueryTags dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY
+AS
+    SET NOCOUNT    ON
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+        DECLARE @studyKey BIGINT
+        DECLARE @seriesKey BIGINT
+        DECLARE @instanceKey BIGINT
+        DECLARE @watermark BIGINT
+
+        -- Add lock so that the instance won't be removed
+        DECLARE @status TINYINT
+        SELECT
+            @studyKey = StudyKey,
```

Comment:
I don't think we use this syntax, do we? Let's use `AS` instead, like `@studyKey AS StudyKey`